From b10081a5176add25e35930acd789921c87545042 Mon Sep 17 00:00:00 2001
From: Chris Packham <judge.packham@gmail.com>
Date: Fri, 14 Oct 2022 21:35:25 +1300
Subject: [PATCH 1/2] zlib: Update to zlib-1.2.13

Brings in the following changes

- Fix a bug when getting a gzip header extra field with
  inflateGetHeader(). This remedies CVE-2022-37434.
- Fix a bug in block type selection when Z_FIXED used. Now the smallest
  block type is selected, for better compression.
- Fix a configure issue that discarded the provided CC definition.
- Correct incorrect inputs provided to the CRC functions. This mitigates
  a bug in Java.
- Repair prototypes and exporting of the new CRC functions.
- Fix inflateBack to detect invalid input with distances too far.

Mark zlib-1.2.12 as obsolete.

Signed-off-by: Chris Packham <judge.packham@gmail.com>
(cherry picked from commit 878a16a13af1024356f2aa623ef7419ec82d4d76)
---
 packages/zlib/1.2.12/version.desc             |  1 +
 .../zlib/1.2.13/0000-mingw-static-only.patch  | 28 +++++++++++++++
 .../0001-crossbuild-macos-libtool.patch       | 34 +++++++++++++++++++
 packages/zlib/1.2.13/chksum                   |  8 +++++
 packages/zlib/1.2.13/version.desc             |  0
 5 files changed, 71 insertions(+)
 create mode 100644 packages/zlib/1.2.13/0000-mingw-static-only.patch
 create mode 100644 packages/zlib/1.2.13/0001-crossbuild-macos-libtool.patch
 create mode 100644 packages/zlib/1.2.13/chksum
 create mode 100644 packages/zlib/1.2.13/version.desc

diff --git a/packages/zlib/1.2.12/version.desc b/packages/zlib/1.2.12/version.desc
index e69de29bb..fcfe38915 100644
--- a/packages/zlib/1.2.12/version.desc
+++ b/packages/zlib/1.2.12/version.desc
@@ -0,0 +1 @@
+obsolete='yes'
diff --git a/packages/zlib/1.2.13/0000-mingw-static-only.patch b/packages/zlib/1.2.13/0000-mingw-static-only.patch
new file mode 100644
index 000000000..cab2389ca
--- /dev/null
+++ b/packages/zlib/1.2.13/0000-mingw-static-only.patch
@@ -0,0 +1,28 @@
+---
+ win32/Makefile.gcc |    8 ++++++--
+ 1 file changed, 6 insertions(+), 2 deletions(-)
+
+--- a/win32/Makefile.gcc
++++ b/win32/Makefile.gcc
+@@ -66,7 +66,11 @@
+        gzwrite.o infback.o inffast.o inflate.o inftrees.o trees.o uncompr.o zutil.o
+ OBJA =
+ 
+-all: $(STATICLIB) $(SHAREDLIB) $(IMPLIB) example.exe minigzip.exe example_d.exe minigzip_d.exe
++all: $(STATICLIB) example.exe minigzip.exe
++
++ifeq ($(SHARED_MODE),1)
++all: $(SHAREDLIB) $(IMPLIB) example_d.exe minigzip_d.exe
++endif
+ 
+ test: example.exe minigzip.exe
+ 	./example
+@@ -120,7 +124,7 @@
+ .PHONY: install uninstall clean
+ 
+ install: zlib.h zconf.h $(STATICLIB) $(IMPLIB)
+-	@if test -z "$(DESTDIR)$(INCLUDE_PATH)" -o -z "$(DESTDIR)$(LIBRARY_PATH)" -o -z "$(DESTDIR)$(BINARY_PATH)"; then \
++	@if test -z "$(INCLUDE_PATH)" -o -z "$(LIBRARY_PATH)" -o -z "$(BINARY_PATH)"; then \
+ 		echo INCLUDE_PATH, LIBRARY_PATH, and BINARY_PATH must be specified; \
+ 		exit 1; \
+ 	fi
diff --git a/packages/zlib/1.2.13/0001-crossbuild-macos-libtool.patch b/packages/zlib/1.2.13/0001-crossbuild-macos-libtool.patch
new file mode 100644
index 000000000..daa624480
--- /dev/null
+++ b/packages/zlib/1.2.13/0001-crossbuild-macos-libtool.patch
@@ -0,0 +1,34 @@
+From: Heiko Lewin <heiko.lewin@worldiety.de>
+Date: Fri, 22 May 2020 03:32:33 +0200
+Subject: configure: use LIBTOOL variable for Darwin builds
+
+---
+ configure |   10 ++++++++--
+ 1 file changed, 8 insertions(+), 2 deletions(-)
+
+--- a/configure
++++ b/configure
+@@ -69,6 +69,12 @@
+ else
+     NM=${NM-"nm"}
+ fi
++if "${CROSS_PREFIX}libtool" --version >/dev/null 2>/dev/null || test $? -lt 126; then
++    LIBTOOL=${LIBTOOL-"${CROSS_PREFIX}libtool"}
++    test -n "${CROSS_PREFIX}" && echo Using ${LIBTOOL} | tee -a configure.log
++else
++    LIBTOOL=${LIBTOOL-"libtool"}
++fi
+ 
+ # set defaults before processing command line options
+ LDCONFIG=${LDCONFIG-"ldconfig"}
+@@ -260,8 +266,8 @@
+         SHAREDLIBV=libz.$VER$shared_ext
+         SHAREDLIBM=libz.$VER1$shared_ext
+         LDSHARED=${LDSHARED-"$cc -dynamiclib -install_name $libdir/$SHAREDLIBM -compatibility_version $VER1 -current_version $VER3"}
+-        if libtool -V 2>&1 | grep Apple > /dev/null; then
+-            AR="libtool"
++        if ${LIBTOOL} -V 2>&1 | grep Apple > /dev/null; then
++            AR="${LIBTOOL}"
+         else
+             AR="/usr/bin/libtool"
+         fi
diff --git a/packages/zlib/1.2.13/chksum b/packages/zlib/1.2.13/chksum
new file mode 100644
index 000000000..214209bd2
--- /dev/null
+++ b/packages/zlib/1.2.13/chksum
@@ -0,0 +1,8 @@
+md5 zlib-1.2.13.tar.xz 7d9fc1d78ae2fa3e84fe98b77d006c63
+sha1 zlib-1.2.13.tar.xz b459b42aa533968ae403c0002a255d62d3136954
+sha256 zlib-1.2.13.tar.xz d14c38e313afc35a9a8760dadf26042f51ea0f5d154b0630a31da0540107fb98
+sha512 zlib-1.2.13.tar.xz 9e7ac71a1824855ae526506883e439456b74ac0b811d54e94f6908249ba8719bec4c8d7672903c5280658b26cb6b5e93ecaaafe5cdc2980c760fa196773f0725
+md5 zlib-1.2.13.tar.gz 9b8aa094c4e5765dabf4da391f00d15c
+sha1 zlib-1.2.13.tar.gz 55eaa84906f31ac20d725aa26cd20839196b6ba6
+sha256 zlib-1.2.13.tar.gz b3a24de97a8fdbc835b9833169501030b8977031bcb54b3b3ac13740f846ab30
+sha512 zlib-1.2.13.tar.gz 99f0e843f52290e6950cc328820c0f322a4d934a504f66c7caa76bd0cc17ece4bf0546424fc95135de85a2656fed5115abb835fd8d8a390d60ffaf946c8887ad
diff --git a/packages/zlib/1.2.13/version.desc b/packages/zlib/1.2.13/version.desc
new file mode 100644
index 000000000..e69de29bb

From e9816b323fee1906e7ed7a6e00b49c69b1fd3bfb Mon Sep 17 00:00:00 2001
From: Chris Packham <judge.packham@gmail.com>
Date: Sun, 16 Oct 2022 13:00:30 +1300
Subject: [PATCH 2/2] zlib: Replace sourceforge mirror with github

New zlib versions aren't being published on sourceforge. Use the github
mirror instead.

Signed-off-by: Chris Packham <judge.packham@gmail.com>
(cherry picked from commit 82346dd7dfe7ed20dc8ec71e193c2d3b1930e22d)
---
 packages/zlib/package.desc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/packages/zlib/package.desc b/packages/zlib/package.desc
index d85e56560..2a3984f3e 100644
--- a/packages/zlib/package.desc
+++ b/packages/zlib/package.desc
@@ -1,4 +1,4 @@
 repository='git https://github.com/madler/zlib.git'
-mirrors='http://downloads.sourceforge.net/project/libpng/zlib/${CT_ZLIB_VERSION} https://www.zlib.net/'
+mirrors='https://github.com/madler/zlib/releases/download/v${CT_ZLIB_VERSION} https://www.zlib.net/'
 archive_formats='.tar.xz .tar.gz'
 signature_format='packed/.asc'
